third_party/nvidia/lib/Dialect/NVWS/IR/Ops.cpp

```diff
@@ -139,4 +140,31 @@ void CreateTokenOp::build(::mlir::OpBuilder &builder,
   build(builder, state, resultType, num, loadType);
 }
 
+static LogicalResult
+verifyBarrierType(Operation *op, mlir::triton::gpu::MemDescType barrierType) {
```

Comment:
Can you factor this with the identical function in `lib/Dialect/TritonNvidiaGPU/IR/Ops.cpp` and reuse it here and there?

Reply:
moved to `ttng` utilities.cpp and call from ttng/nvws dialects